lua/lvim/config/_deprecated.lua

```diff
@@ -138,34 +144,39 @@ function M.post_load()
     for old_key, alternative in pairs(alternatives) do
       if spec[old_key] ~= nil then
         local message
+        local old_value = vim.inspect(spec[old_key]) or "value"
 
         if type(alternative) == "function" then
           message = alternative()
         else
           spec[alternative] = spec[old_key]
         end
-        spec[old_key] = nil
-
-        local new_value = vim.inspect(spec[alternative] or "[value]")
-        message = message or string.format("%s = %s", alternative, new_value)
-        vim.schedule(function()
-          vim.notify_once(
-            string.format(
-              [[`%s` in `lvim.plugins` has been deprecated since the migration to lazy.nvim. Use `%s` instead.
+
+        if type(alternative) ~= "function" or message then
```

Comment:
`message` would only have a value if `type(alternative) == "function"` so either this condition is redundant, or you meant to put it after this assignment https://github.com/LunarVim/LunarVim/blob/816a6b2a280dab5ad0401804e0752eae2667e40e/lua/lvim/config/_deprecated.lua#L159

Reply:
in this new case (tag ~= "*") the type would be function and message would be nil so the deprecation message shouldn't be shown

Reply:
it can be set here https://github.com/LunarVim/LunarVim/blob/816a6b2a280dab5ad0401804e0752eae2667e40e/lua/lvim/config/_deprecated.lua#L150

Reply:
> it can be set here
> 
> https://github.com/LunarVim/LunarVim/blob/816a6b2a280dab5ad0401804e0752eae2667e40e/lua/lvim/config/_deprecated.lua#L150

which can only be reached when ` type(alternative) == "function"`! 

here's what that block is doing (renamed `message` to make it clearer)

```lua
local is_function

if type(alternative) == "function" then
  is_function = alternative()
--[...]
end

if type(alternative) ~= "function" or is_function then -- this condition is always true :)
 -- [ ... ]
end
```

Reply:
it's not always true, this only returns a value if tag is equal to "*"
```lua
    alternatives.tag = function()
      if spec.tag == "*" then
        spec.version = "*"
        return [[version = "*"]]
      end
    end
```

Reply:
ooh, things make more sense now! 

~~I wonder if we shouldn't just offload this to `lazy` it should complain about an invalid `tag`, and it's not like `tag = "*"` was that common in the first place.~~ _it doesn't seem that `lazy` is doing any validation for that.._

---

maybe add a comment for now just so we won't forget about it, and hopefully we would remove all of this handling in the next release.

Reply:
done